File: frontend/src/api/reviewApi.ts
import { ReviewResponse } from '../types/Review';
import axiosClient from './axiosClient.ts';

interface CreateReviewData {
  orderId: string;
  productId: string;
  rating: number;
  comment: string;
}

export const reviewApi = {
  getReviewsByProduct: (productId: string): Promise<ReviewResponse> => {
    return axiosClient.get(`/products/${productId}/reviews`);
  },

  createReview: (data: CreateReviewData): Promise<any> => {
    return axiosClient.post(`/reviews`, data);
  }
};

File: Mobileptit/app/src/main/java/com/uteshop/app/data/remote/api/ReviewApi.java
package com.uteshop.app.data.remote.api;

import com.uteshop.app.data.remote.dto.request.CreateReviewRequest;
import com.uteshop.app.data.remote.dto.response.*;
import retrofit2.Call;
import retrofit2.http.*;

public interface ReviewApi {
    @GET("v1/api/products/{productId}/reviews")
    Call<ReviewListResponse> getReviewsByProduct(@Path("productId") String productId);

    @GET("v1/api/reviews")
    Call<ReviewListResponse> getMyReviews();

    @POST("v1/api/reviews")
    Call<CreateReviewResponse> createReview(@Body CreateReviewRequest r);
}
